### PR TITLE
Some thread-pool-executor parallelizations, part 1/n

### PR DESCRIPTION
**Pull request by johnkerl**
Several more callsites can benefit from this pattern

### CODE REVIEW

File: apis/python/src/tiledbsc/annotation_dataframe.py [hunk 1]
```diff
@@ -186,14 +187,29 @@ def query(
...
         """
-        for k in df:
-            dfk = df[k]
-            if len(dfk) > 0 and type(dfk[0]) == bytes:
-                df[k] = dfk.map(lambda e: e.decode())
+        futures = []
+        with ThreadPoolExecutor() as executor:
+            for k in df:
+                future = executor.submit(
+                    self._ascii_to_unicode_series_readback, k, df[k]
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsc/annotation_dataframe.py:204]: Is this a pure python function? Does `dfk.map` hold the GIL?

**Review by Shelnutt2 — APPROVED** [review]

File: apis/python/src/tiledbsc/annotation_dataframe.py (hunk 1 above)
Comment (johnkerl) [reply apis/python/src/tiledbsc/annotation_dataframe.py:204]: @Shelnutt2 I have much to learn -- will try to find out

Comment (johnkerl) [reply apis/python/src/tiledbsc/annotation_dataframe.py:204]: I did put in timings & this went from 1.7 seconds to 0.8 on a particular array

**Comment by johnkerl** [comment]
C++ build failing, rest intact -- merging pending analysis of the separate C++ issue not touched by this PR.

**Pull request merged**